server/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -149,11 +149,8 @@ void addIndices(final RecoveryState.Index indexRecoveryStats, final Directory ta
             final long maxSeqNo, final long maxUnsafeAutoIdTimestamp, IndexMetaData indexMetaData, int shardId, boolean split,
             boolean hasNested) throws IOException {
 
-        // clean target directory (if previous recovery attempt failed) and create a fresh segment file with the proper lucene version
-        Lucene.cleanLuceneIndex(target);
         assert sources.length > 0;
         final int luceneIndexCreatedVersionMajor = Lucene.readSegmentInfos(sources[0]).getIndexCreatedVersionMajor();
```

Comment:
maybe for uniformity pass the lucene version as parameter to the `addIndices` method and use `indexShard.indexSettings().getIndexVersionCreated().luceneVersion`, just as in all the other places?

Reply:
I felt like reusing the version of one index was safer, as the Lucene version is sometimes guessed via logic in o.e.Version. I don't feel strongly about it though so if you'd rather like that we use the version that comes from index settings, I'm fine with it.

Reply:
ok, thanks for explaining your reasoning here. Let's keep as is, then.